### PR TITLE
(hopefully) fixed duplicate name exploit when creating a character

### PR DESCRIPTION
**Pull request by Z1Meme**

### CODE REVIEW

**Comment by Z1Meme** [comment]
@QuentinGruber I was playing my server and someone hopped on it with my same name, didn't know we didn't check this lol

File: src/servers/LoginServer/loginserver.ts [hunk 1]
```diff
@@ -450,16 +450,25 @@ export class LoginServer extends EventEmitter {
...
   }
 
-  TunnelAppPacketClientToServer(client: Client, packet: any) {
+  async TunnelAppPacketClientToServer(client: Client, packet: any) {
     const baseResponse = { serverId: packet.serverId };
     let response;
     switch (packet.subPacketName) {
       case "nameValidationRequest":
+        let status = 1;
+        if(this._soloMode) {
```
Comment (QuentinGruber) [inline src/servers/LoginServer/loginserver.ts:459]:
```suggestion
        if(!this._soloMode) {
```

File: src/servers/LoginServer/loginserver.ts [hunk 2]
```diff
@@ -450,16 +450,25 @@ export class LoginServer extends EventEmitter {
...
+          const duplicateCharacter = await this._db
+          .collection("characters-light")
+          .findOne({ characterName: packet.result.characterName });
```
Comment (QuentinGruber) [inline src/servers/LoginServer/loginserver.ts:462]:
```suggestion
          .findOne({ characterName: packet.result.characterName, serverId: baseResponse.serverId });
```

**Review by QuentinGruber — CHANGES_REQUESTED** [review]

**Comment by QuentinGruber** [comment]
> @QuentinGruber I was playing my server and someone hopped on it with my same name, didn't know we didn't check this lol

There was an old implementation on live-login-server, i will removed it and add also the blackListEntries verification on this PR, you can check it there https://github.com/H1emu/live-login-server

File: src/servers/LoginServer/loginserver.ts (hunk 2 above)
Comment (QuentinGruber) [reply src/servers/LoginServer/loginserver.ts:462]: This still need to be done @Z1Meme

**Review by QuentinGruber — APPROVED** [review]

**Pull request merged**